astra/src/main/java/com/slack/astra/writer/SpanFormatter.java

```diff
@@ -148,14 +149,24 @@ public static List<Trace.KeyValue> convertKVtoProto(
   @VisibleForTesting
   public static List<Trace.KeyValue> convertKVtoProtoDefault(
       String key, Object value, Schema.IngestSchema schema) {
+    return convertKVtoProtoDefault(key, value, schema, DEPTH_LIMIT - 1);
+  }
+
+  @VisibleForTesting
+  public static List<Trace.KeyValue> convertKVtoProtoDefault(
+      String key, Object value, Schema.IngestSchema schema, int depthLimit) {
     List<Trace.KeyValue> tags = new ArrayList<>();
     if (value instanceof Map) {
-      // todo - consider adding a depth param to prevent excessively nested fields
+      if (depthLimit <= 0) {
+        return tags;
```

Comment:
Instead of returning empty list of tags, can we instead return a string blob that is un-indexed? Since logging systems are often used as debugging systems, it would be ideal to ingest data where possible.

Reply:
argh. Yeah. We'll need to do that now, but it'll be sort of broken. I'd assumed that the _source field would retain the original json, but this change here: https://github.com/slackhq/astra/pull/816/files#diff-3e3f1e6e9032267dd01c35b0f82f031bd633e1f08355002149316079bbe5711b broken that. It got rid of the code that reified the original document in favor of flattening everything. I'm not sure it was an intended because it isn't mentioned in the PR and there are no tests that cover the change.

Kinda like this:
Before:
`{"a":{"b":1}}` => `{"a.b":1,"_source":{"a":{"b":1}}}`
After:
`{"a":{"b":1}}` => `{"a.b":1,"_source":{"a.b":1}}`

Reply:
Hm. Maybe not exactly that PR, but certainly from that series of changes.

Reply:
>  can we instead return a string blob that is un-indexed?
Yes. I'll mark it as BINARY, which will cause it to be unindexed but still be injected into the document correctly.